fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/IterableHelpers.java

```diff
@@ -56,6 +56,23 @@ private IterableHelpers() {
                 }).iterator();
     }
 
+    /**
+     * Maps given function to the given iterable and returns a new iterable. This is akin
+     * to compositions based on {@link java.util.stream.Stream}, however, allows for restarting the iteration as per
+     * general contract in {@link Iterable}.
+     * @param source source iterable
+     * @param mapper mapper function to map from {@code T} to {@code Iterable} of {@code R}
+     * @param <T> type of source iterable
+     * @param <R> type the given mapper function returns
+     * @return an iterable of type {@code R} that is the conceptual map of {@code source} using {@code mapper}
+     */
+    public static <T, R> Iterable<R> map(@Nonnull Iterable<T> source, @Nonnull final Function<? super T, R> mapper) {
```

Comment:
I believe that this is very close to `Iterables.transform` in Guava. We already have that as a dependency, but perhaps it isn't used much in this part of the system.

Reply:
Using that Guava method causes a code check warning `StaticPseudoFunctionalStyleMethod` as people need to be told to use streams. Since I need something like _restartable_ streams I went with iterators. As a lot of callers already use `flatMap()` from this class, I think I want to keep using `map()` from here as well. I looked into the `transform()` implementation in guava which does not use streams under the cover but iterators directly. I think `map()` as it is written here is actually nicer. I don't have strong opinions here but I will leave this as is.

Reply:
If `StaticPseudoFunctionalStyleMethod` were smarter, wouldn't it complain about this method? The claim seems to be that one should use streams in the caller and not in yet another static monad-like method.